powersimdata/input/profile_input.py

```diff
@@ -68,8 +78,10 @@ def _read(self, f, path):
         data = pd.read_csv(f, index_col=0, parse_dates=True)
         if "demand_flexibility" in path:
             data.columns = data.columns.astype(str)
-        else:
+        elif all(c.isdigit() for c in data.columns):
             data.columns = data.columns.astype(int)
+        else:
+            data.columns = data.columns.astype(str)
```

Comment:
I am surprised that `pandas` is not able to assign the correct type to column's header